pkg/tools/list/graph.go

```diff
@@ -141,17 +122,30 @@ func (g *Graph) DeleteVertex(id string) error {
 
 	// delete the vertex from the vertices map
 	g.vertices.Remove(id)
+
+	// delete the edge from the edges map
+	for edge := range g.edges {
+		if edge.Source().Id() == id || edge.Target().Id() == id {
+			delete(g.edges, edge)
```

Comment:
do not delete map item in for range map, it is danger.

Reply:
I used to think so too but after some digging, I found it safe: Although in many languages deleting while iterating is dangerous, it's safe in Go. 
ref: 
- https://go.dev/ref/spec#For_statements
- https://stackoverflow.com/questions/23229975/is-it-safe-to-remove-selected-keys-from-map-within-a-range-loop